modules/core/src/main/kotlin/datamaintain/core/step/check/rules/contracts/FullContextCheckRule.kt

```diff
@@ -1,11 +1,16 @@
 package datamaintain.core.step.check.rules.contracts
 
+import datamaintain.core.exception.DatamaintainCheckException
 import datamaintain.core.script.ExecutedScript
 import datamaintain.core.script.ScriptWithContent
 import datamaintain.core.step.check.rules.CheckRule
 
 abstract class FullContextCheckRule(
         val executedScripts: Sequence<ExecutedScript>
 ): CheckRule {
+    /**
+     * @throws DatamaintainCheckException
+     */
+    @Throws(DatamaintainCheckException::class)
```

Comment:
I am wondering if it is really needed to annotate the method. It is only Java compatibility concern. As far as I know this api will not be exposed outside of Datamaintain.

Reply:
I think it is a good reminder for users extending the class in Java for implement a custom check, even if throw DatamaintainCheckException is not mandatory.